src/utransport.rs

```diff
@@ -482,7 +482,13 @@ impl UTransport for UPClientZenoh {
                         )
                     })?;
                 // Get Zenoh key
-                let topic = attributes.clone().source;
+                let topic = if attributes.sink.is_some() {
```

Comment:
Yup, seems like this is what we need :slightly_smiling_face: 

Just wanna get some input from @sophokles73

Reply:
It is correct to assume that you are dealing with a *Notification* if *sink* is not empty and a *Publish* otherwise.